src/Client.php

```diff
@@ -51,391 +52,164 @@ class Client implements ClientInterface
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getRepresentationSerializer()
-    {
-        return $this->representationSerializer;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function setRepresentationSerializer(ReprSerializer $representationSerializer)
-    {
-        $this->representationSerializer = $representationSerializer;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getSerializer()
-    {
-        return $this->serializer;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function setSerializer(Serializer $serializer)
-    {
-        $this->serializer = $serializer;
-    }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function captureMessage(string $message, array $params = [], array $payload = [])
-    {
-        $payload['message'] = $message;
-        $payload['message_params'] = $params;
 
-        return $this->capture($payload);
+        $this->installedIntegrations = Handler::setupIntegrations($integrations);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function captureException(\Throwable $exception, array $payload = [])
+    public function getOptions(): Options
     {
-        $payload['exception'] = $exception;
-
-        return $this->capture($payload);
+        return $this->options;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function captureLastError(array $payload = [])
+    public function addBreadcrumb(Breadcrumb $breadcrumb, ?Scope $scope = null): void
     {
-        $error = error_get_last();
+        $beforeBreadcrumbCallback = $this->options->getBeforeBreadcrumbCallback();
+        $maxBreadcrumbs = $this->options->getMaxBreadcrumbs();
 
-        if (null === $error || !isset($error['message'][0])) {
-            return null;
+        if ($maxBreadcrumbs <= 0) {
+            return;
         }
 
-        $exception = new \ErrorException(@$error['message'], 0, @$error['type'], @$error['file'], @$error['line']);
-
-        return $this->captureException($exception, $payload);
-    }
+        $breadcrumb = $beforeBreadcrumbCallback($breadcrumb);
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getLastEvent()
-    {
-        return $this->lastEvent;
+        if (null !== $breadcrumb && null !== $scope) {
+            $scope->addBreadcrumb($breadcrumb, $maxBreadcrumbs);
+        }
     }
 
     /**
-     * {@inheritdoc}
+     * Assembles an event and prepares it to be sent of to Sentry.
+     *
+     * @param array      $payload
+     * @param null|Scope $scope
+     *
+     * @return null|Event
      */
-    public function getLastEventId()
+    protected function prepareEvent(array $payload, ?Scope $scope = null): ?Event
     {
-        @trigger_error(sprintf('The %s() method is deprecated since version 2.0. Use getLastEvent() instead.', __METHOD__), E_USER_DEPRECATED);
-
-        if (null === $this->lastEvent) {
+        if (mt_rand(1, 100) / 100.0 > $this->getOptions()->getSampleRate()) {
```

Comment:
>even though just discarding random events seems really wrong experience showed that it still works surprisingly well

I can't see any way this can work surprisingly well :open_mouth: I mean, you're discarding events in a totally random order and you could miss some important events. Imho you should not provide this feature because it encourages a bad habit